whitesource/client.py

```diff
@@ -79,7 +79,7 @@ def save_project_tag(self, projectToken: str, key: str, value: str):
 
     async def upload_to_project(
         self,
-        extra_whitesource_config: typing.Dict,
+        extra_whitesource_config: typing.Union[None, dict],
```

Comment:
Didn't we agree on `typing.Optional[dict]` for these cases?

But in this case (a simple function def) I'd say `extra_whitesource_config: dict = None,` is sufficient.

In general (in our codebase at least), these hints are necessary for dataclasses to work correctly. For all other cases, they are merely hints to the developer. As they have to look up the function definition either way, I'd argue to use the "standard" way of declaring an optional variable. Even though the typehint should still technically be `typing.Optional[…]` I think readability is better with just the plain typehint+default.

Similar on other appearances of this kind of annotations on functions.

Reply:
Union[xxx, None] is equivalent to Optional[xxx] @AndreasBurger 
I agree that the Optional-shorthand is the better choice, though

Reply:
Yes, we did agree on `typing.optional` @AndreasBurger , these hints just sneaked through.
Will go for the default value approach since its more common in our codebase.